tests/unit/wordpress_mock.py

```diff
@@ -0,0 +1,526 @@
+        self._databases[key] = None
+        self._database_credentials[key] = {"user": user, "password": password}
+
+    def database_can_connect(self, host: str, database: str) -> bool:
+        """Test if given host and database can connect to a simulated database.
+
+        Args:
+            host: database host.
+            database: database name.
+        """
+        key = self._database_identifier(host, database)
+        return key in self._databases
+
+    def database_can_login(self, host: str, database: str, user: str, password: str) -> bool:
+        """Test if given database credentials can connect to a simulated database.
+
+        Args:
+            host: database host.
+            database: database name.
+            user: database user.
+            password: database password.
+        """
+        key = self._database_identifier(host, database)
+        if key not in self._database_credentials:
+            raise KeyError(f"Database (host={host}, database={database}) does not exist")
+        credential = self._database_credentials[key]
+        return credential["user"] == user and credential["password"] == password
+
+    def install_wordpress(self, host: str, database: str) -> None:
+        """Install WordPress on a simulated database.
+
+        Args:
+            host: database host.
+            database: database name.
+        """
+        key = self._database_identifier(host, database)
+        if key not in self._databases:
+            raise KeyError(f"Database (host={host}, database={database}) does not exist")
+        if self._databases[key] is not None:
+            raise KeyError(f"Wordpress already installed on (host={host}, database={database}).")
+        self._databases[key] = WordPressDatabaseInstanceMock(
+            builtin_options=self._builtin_wordpress_options
+        )
+
+    def is_wordpress_installed(self, host: str, database: str) -> bool:
+        """Test if WordPress is installed on the given simulated database.
+
+        Args:
+            host: database host.
+            database: database name.
+        """
+        key = self._database_identifier(host, database)
+        if key not in self._databases:
+            raise KeyError(f"Database (host={host}, database={database}) does not exist")
+        return self._databases[key] is not None
+
+    def get_wordpress_database(self, host, database) -> WordPressDatabaseInstanceMock:
+        """Get the simulated WordPress installed database.
+
+        Args:
+            host: database host.
+            database: database name.
+        """
+        key = self._database_identifier(host, database)
+        if not self.is_wordpress_installed(host, database):
+            raise KeyError(f"Wordpress isn't installed on (host={host}, database={database}).")
+        return self._databases[key]
+
+
+class MysqlConnectorMock:
+    """A mock for :py:mod:`mysql.connector`."""
+
+    Error = mysql.connector.Error
```

Comment:
I'm curios for the reason to store the error class on the class please?

Reply:
This class is a mocking the `mysql.connector` module, so `MysqlConnectorMock.Error` should map to  `mysql.connector. Error`.

Reply:
Ah I see, do you think the above as a comment on this line would help future readers of the code understand that too?

Reply:
Added, thanks!